doc/fgpu.design.rst

```diff
@@ -259,14 +339,121 @@ written as :math:`pm + s`. Let :math:`z^r` denote the array :math:`z_r, z_{n+r},
 The whole expression is a Fourier transform of the expression in brackets
 (the exponential inside the bracket is the so-called "twiddle factor").
 
-An inconvenience of this structure is that :math:`z^r` is not a contiguous
-set of input samples, but a strided array. While cuFFT does support both
-strided inputs and batched transformations, we cannot batch over :math:`r`
-and over multiple spectra at the same time as it only supports a single batch
-dimension with corresponding stride. We solve this by modifying the PFB kernel
-to reorder its output such that each :math:`z^r` is output contiguously. This
-can be done by shuffling some bits in the output index (because we assume
-powers of two everywhere).
+Now let's walk through each part. To recap the indexing used in the Cooley-Tukey algorithm: let a
+time-domain index :math:`i` be written as :math:`qn + r` and a frequency-domain index :math:`k` be
+written as :math:`pm + s`. Let :math:`z^r` denote the array :math:`z_r, z_{n+r}, \dots, z_{(m-1)n+r}`,
+and denote its Fourier transform by :math:`Z^r`. It is worthwhile to point out that the superscript
+:math:`r` *does not* denote raise to the power but rather is a means to indicate an :math:`r^{th}` array.
+In practice this :math:`r^{th}` array is a subset (part) of the larger :math:`z` array of input data.
+
+Let's unpack this a bit further --- what is actually happening
+is that the initial array :math:`z` is divided into :math:`n=4` separate arrays each of
+:math:`m=32768/4 = 8192` elements (hence the :math:`N = mn` above)(This is true for a desired overall
+transform size of 32768 channels). The actual samples that land up in each array are defined by the
+indices :math:`i` and :math:`k`. Lets start with :math:`i`.
```

Comment:
```suggestion
indices :math:`i` and :math:`k`. Let's start with :math:`i`.
```

Reply:
Ending a paragraph "Let's start with i" is a bit of a cliff-hanger. It might be more sensible to use that as the start of the next paragraph.